.travis.yml

```diff
@@ -5,16 +5,28 @@ go_import_path: github.com/pingcap/tidb
 go:
   - "1.11.3"
 
-env:
-    - TRAVIS_COVERAGE=0
-    - TRAVIS_COVERAGE=1
+cache:
+  directories:
+    - $GOPATH/pkg/mod
+  # TODO: If space permits, we could cache this dirctory.
+  # - $HOME/.cache/go-build
+
+env: 
+    - TRAVIS_COVERAGE=0 TEST_SUITE=check
+    - TRAVIS_COVERAGE=0 TEST_SUITE=checkdep
+    - TRAVIS_COVERAGE=0 TEST_SUITE=gotest
+    - TRAVIS_COVERAGE=0 TEST_SUITE=explaintest
+    - TRAVIS_COVERAGE=1 TEST_SUITE="dev upload-coverage"
```

Comment:
If you think that I spilled it too trivial so that we take much time to boot VM.
We could merge something later.